openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -603,6 +625,141 @@ Location: https://client.example.net/cb?
+
+ - In case a wallet attestation is required by the Issuer, it has to be included in this request.
+ - In case the Wallet has received an `auth_session` parameter previously, it has to be included in this request (see (#error-request-presentation)).
+ - In case the Wallet has completed a Presentation, it has to include the received redirect URI in the parameter `openid4vp_redirect_uri` (see (#error-request-presentation)) during the next call to the Authorization Challenge Endpoint.
+
+## Authorization Challenge Response
+
+Upon receiving an Authorization Challenge Request, the Issuer determines whether the information provided by the Wallet so far is sufficient to grant authorization for the Credential issuance. 
+
+Depending on this assessment, the response from the Authorization Challenge Endpoint can take one of the following forms:
+
+### Error - Request Presentation {#error-request-presentation}
+
+The Authorization Server MAY request a Credential Presentation by responding with an error response, indicated by an HTTP 400 (Bad Request) status code. The content type of the error response MUST be `application/json` and the JSON document in the body MUST indicate the `error` code `insufficient_authorization` as shown in the following example. In this case, the response MUST include an `openid4vp_presentation` parameter containing an Authorization Request as defined in Section 5 of [@!OpenID4VP], encoded in JSON.
+
+```
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "error": "insufficient_authorization",
+  "auth_session": "123456789",
+  "openid4vp_presentation": {
+    "client_id": "x509_san_dns:rp.example.com",
+    "request_uri": "https://rp.example.com/oidc/request/1234"
+  }
+}
+```
+
+The Authorization Server MAY use the `auth_session` parameter in the Authorization Challenge Response. The auth session allows the Authorization Server to associate subsequent requests by this Wallet with the ongoing authorization request sequence. The Wallet MUST include the `auth_session` in follow-up requests to the Authorization Challenge Endpoint if it receives one along with the error response.
+
+If the error code is `insufficient_authorization` and an `openid4vp_presentation` parameter is present, the Wallet MUST process the Authorization Request contained in the `openid4vp_presentation` parameter as defined in [@!OpenID4VP] to perform a Credential Presentation to the Authorization Server.
+
+For the requested Presentation, the Issuer is acting as a Verifier to the Wallet. The exact architecture and the deployment of the Issuer's OpenID4VP Verifier is out of scope of this specification.
+
+In the Authorization Request contained in the `openid4vp_presentation` parameter, the `response_type` MUST be `vp_token` (defined in Section 8 of [@!OpenID4VP]).
+
+The `response_mode` of the request MUST be `direct_post` (defined in Section 8.2 of [@!OpenID4VP]).
```

Comment:
I can't decide if this is overly restrictive. direct_post.jwt would I think be fine too, and some people seem to prefer things are encrypted.

```suggestion
The `response_mode` of the request MUST be `direct_post` (defined in Section 8.2 of [@!OpenID4VP]) or `direct_post.jwt` (defined in Section 8.3.1 of [@!OpenID4VP]).
```